setup.py

```diff
@@ -1,55 +1,65 @@
-from setuptools import setup, find_packages
-
 import codecs
 import os
 import re
-import sys
+
+from setuptools import setup, find_packages
 
 here = os.path.abspath(os.path.dirname(__file__))
 
 
 min_requires = [
-    'google-auth-httplib2',
-    'google-auth',
-    'google-cloud-core>=1.4.1',
-    'google-cloud-storage',
-    'requests',
-    'tqdm',
-    'urllib3',
-    'deprecated',
-    'python-dotenv',
-    'gcsfs>=0.3.0,<0.7',
-    "retry"
+    "deprecated",
+    "gcsfs>=0.3.0,<0.7",
+    "google-auth",
+    "google-auth-httplib2",
+    "google-cloud-bigquery",
+    "google-cloud-core>=1.4.1",
+    "google-cloud-storage",
+    "joblib>=0.11",
+    "numpy>=1.16.3",
+    "pandas>=0.23.4,!=1.0.4",
```

Comment:
do we want to have these dependencies default? 

We isolated the min requirement because we didn't want to have heavy dependencies in pycarol. For those that only want to call the APIs. This was important for the assistant. 

What should we do?

Reply:
good point. It seems to me that since bigquery will be the default way to query data from staging and models, and with the possibility of ES being dropped, it seems that pandas and numpy are almost natural dependencies.
Another point is how much boilerplate the code has to deal with an optional pandas presence? It seems that we could simplify the code a lot by assuming these dependencies. For luigi or online apps it seems to make more sense, since there are lots of batch apps with no flask requirement or even luigi. The same applies to dask.
Concerning assistant: how heavy is to have those dataframe dependencies? They are kind of lightweight for the docker image, right (some hundreds of MB top I guess).

Reply:
I also think both will always be used. 

@gzamboni @pedrobuzzi 
Could this be a problem add pandas/numpy as pycarol requeiment? 

I remember there was something when using python inside the assistant flow or something like that..

Reply:
I went back to past min requirements so that it has no impact on assistant. We can discuss that later for a future PR.